src/llmq/quorums_instantsend.cpp

```diff
@@ -837,12 +838,65 @@ void CInstantSendManager::SyncTransaction(const CTransaction& tx, const CBlockIn
     }
 
     bool chainlocked = pindex && chainLocksHandler->HasChainLock(pindex->nHeight, pindex->GetBlockHash());
-    if (!islockHash.IsNull() || chainlocked) {
-        LOCK(cs);
-        pendingRetryTxs.emplace(tx.GetHash());
-    } else {
+    if (islockHash.IsNull() && !chainlocked) {
         ProcessTx(tx, Params().GetConsensus());
     }
+
+    LOCK(cs);
+    if (!chainlocked && islockHash.IsNull()) {
+        // TX is not locked, so make sure it is tracked
+        AddNonLockedTx(MakeTransactionRef(tx));
+        nonLockedTxs.at(tx.GetHash()).pindexMined = posInBlock == CMainSignals::SYNC_TRANSACTION_NOT_IN_BLOCK ? pindex : nullptr;
+    } else {
+        // TX is locked, so make sure we don't track it anymore
+        RemoveNonLockedTx(tx.GetHash());
+    }
+}
+
+void CInstantSendManager::AddNonLockedTx(const CTransactionRef& tx)
+{
+    AssertLockHeld(cs);
+    auto it = nonLockedTxs.find(tx->GetHash());
+    if (it == nonLockedTxs.end()) {
+        it = nonLockedTxs.emplace(tx->GetHash(), NonLockedTxInfo()).first;
```

Comment:
nit: `emplace` should be able to handle both cases (found/not found), no need to use `find`/`end`.